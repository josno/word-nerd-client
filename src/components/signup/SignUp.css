.form {
	margin-left: auto;
	margin-right: auto;
	border: 2px solid black;
	height: fit-content;
	width: 50vw;
}

.form-tab,
.input-submit-button {
	list-style: none;
	display: inline-block;
	padding: 10px;
	border: 2px solid black;
	margin: 10px;
	font-size: 30px;
}

ul {
	padding: 0px;
}

.form-input {
	margin-left: auto;
	margin-right: auto;
	height: 5vh;
	width: 30vw;
	margin: 10px;
	font-size: 26px;
	padding: 0.4rem;
}

a {
	text-decoration: none;
}

.sign-up-sucess-message {
	color: green;
}
