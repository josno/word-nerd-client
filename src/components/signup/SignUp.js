import React, { Component } from 'react';
import './SignUp.css';
import { Link } from 'react-router-dom';
import AuthApiService from '../../services/auth-api-service';

class SignUp extends Component {
	constructor(props) {
		super(props);
		this.state = {
			error: null
		};
	}

	handleSignUp = e => {
		e.preventDefault();
		this.setState({ error: null });
		const { username, password, fullName } = e.target;
		const newUser = {
			user_name: username.value,
			password: password.value,
			full_name: fullName.value
		};

		// console.log(newUser);

		AuthApiService.postUser(newUser)
			.then(res =>
				!res.ok ? res.json().then(e => Promise.reject(e)) : res.json()
			)
			.then(res => console.log(res));
		// .then(user => {
		// 	this.props.history.push('/game-home-page');
		// })
		// .catch(res => {
		// 	this.setState({ error: res.error });
		// });
	};

	capitalizeNames = string => {
		return string.charAt(0).toUpperCase() + string.slice(1);
	};

	render() {
		return (
			<div className="sign-up">
				<form onSubmit={this.handleSignUp}>
					<div className="input-form">
						<input
							className="form-input"
							type="text"
							htmlFor="full-name"
							placeholder="Full Name"
							name="fullName"
						/>

						<input
							className="form-input"
							type="text"
							htmlFor="username"
							placeholder="username"
							name="username"
						/>
						<input
							className="form-input"
							type="password"
							htmlFor="password"
							placeholder="password"
							name="password"
						/>
					</div>
					<button type="submit" className="input-submit-button">
						Sign Up
					</button>
				</form>
			</div>
		);
	}
}

export default SignUp;
