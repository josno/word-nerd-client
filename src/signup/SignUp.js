import React, { Component } from 'react';
import './SignUp.css';

class SignUp extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}
	render() {
		return (
			<div className="sign-up">
				<section>
					<div class="form">
						<ul>
							<li class="form-tab">
								<a href="one-home-page-sign-up.html">Sign Up</a>
							</li>

							<li class="form-tab">
								<a href="one-home-page-login.html">Log In</a>
							</li>
						</ul>
						<form>
							<div class="input-form">
								<input
									class="form-input"
									type="text"
									for="first-name"
									placeholder="First Name"
								/>
								<input
									class="form-input"
									type="text"
									for="last-name"
									placeholder="Last Name"
								/>
								<input
									class="form-input"
									type="text"
									for="username"
									placeholder="username"
								/>
								<input
									class="form-input"
									type="text"
									for="password"
									placeholder="password"
								/>
							</div>
						</form>
						<a href="two-game-home-page.html">
							<button class="input-submit-button">
								Sign Up For Free
							</button>
						</a>
					</div>
				</section>
			</div>
		);
	}
}

export default SignUp;
