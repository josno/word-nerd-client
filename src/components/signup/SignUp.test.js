import React from 'react';
import ReactDOM from 'react-dom';
import SignUp from './SignUp';
import renderer from 'react-test-renderer';

describe('<SignUp/>', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<SignUp />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	it('renders as expected', () => {
		const tree = renderer.create(<SignUp />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
