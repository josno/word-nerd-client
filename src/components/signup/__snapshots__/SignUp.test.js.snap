// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp/> renders as expected 1`] = `
<div
  className="sign-up"
>
  <section>
    <div
      className="input-form"
    >
      <input
        className="form-input"
        htmlFor="first-name"
        placeholder="First Name"
        type="text"
      />
      <input
        className="form-input"
        htmlFor="last-name"
        placeholder="Last Name"
        type="text"
      />
      <input
        className="form-input"
        htmlFor="username"
        placeholder="username"
        type="text"
      />
      <input
        className="form-input"
        htmlFor="password"
        placeholder="password"
        type="text"
      />
    </div>
    <a
      href="two-game-home-page.html"
    >
      <button
        className="input-submit-button"
      >
        Sign Up For Free
      </button>
    </a>
  </section>
</div>
`;
